src/metatensor/models/experimental/pet/train.py

```diff
@@ -0,0 +1,142 @@
+import logging
+from pathlib import Path
+from typing import Dict, List, Optional, Union
+
+import torch
+from metatensor.learn.data import DataLoader
+from metatensor.learn.data.dataset import _BaseDataset
+from metatensor.torch.atomistic import ModelCapabilities
+from pet.hypers import Hypers
+from pet.pet import PET
+from pet.train_model import fit_pet
+
+from ...utils.data import collate_fn
+from ...utils.data.system_to_ase import system_to_ase
+from .model import DEFAULT_HYPERS, Model
+
+
+logger = logging.getLogger(__name__)
+
+
+def train(
+    train_datasets: List[Union[_BaseDataset, torch.utils.data.Subset]],
+    validation_datasets: List[Union[_BaseDataset, torch.utils.data.Subset]],
+    requested_capabilities: ModelCapabilities,
+    hypers: Dict = DEFAULT_HYPERS,
+    continue_from: Optional[str] = None,
+    output_dir: str = ".",
+    device_str: str = "cpu",
+):
+    if torch.get_default_dtype() != torch.float32:
+        raise ValueError("PET only supports float32")
+    if device_str != "cuda" and device_str != "gpu":
+        raise ValueError("PET only supports cuda (gpu) training")
+    if len(requested_capabilities.outputs) != 1:
+        raise ValueError("PET only supports a single output")
+    target_name = next(iter(requested_capabilities.outputs.keys()))
+    if requested_capabilities.outputs[target_name].quantity != "energy":
+        raise ValueError("PET only supports energies as output")
+    if requested_capabilities.outputs[target_name].per_atom:
+        raise ValueError("PET does not support per-atom energies")
+    if len(train_datasets) != 1:
+        raise ValueError("PET only supports a single training dataset")
+    if len(validation_datasets) != 1:
+        raise ValueError("PET only supports a single validation dataset")
+
+    if device_str == "gpu":
```

Comment:
this should never arrive to a model. IMO we should convert the device to a torch device before passing it to the train script

Reply:
You're probably right... or maybe a list of devices. We can open an issue for this, because it concerns all the models